azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -130,7 +130,11 @@ static void setupWatch(final Map<String, ParseConfigFile> configFileMap) throws
             // Match!
             // reparse the config file
             log.info("Modification detected, reloading config file " + filename);
-            configFileMap.get(filename).parseConfigFile();
+            try {
+              configFileMap.get(filename).parseConfigFile();
+            } catch (Exception e) {
```

Comment:
This try-catch should be here so that watcher thread doesn't just exit in case of an exception.

Reply:
Hi, Thanks for suggesting the fix for the test. I already have a patch which fixes the problem, however, I agree that using a try catch block helps keep the thread alive. 
I will update my fix.
https://github.com/azkaban/azkaban/pull/2218